### ISSUE TITLE
runtime error: unsigned integer overflow: 50331649 - 50331650 cannot be represented in type 'unsigned int' (ssl/s3_lib.c:2796:19)

### ISSUE THREAD

**Issue by geeknik** [openssl/openssl#3889]
Similar to #3885, built on a fresh Debian 9 x64 machine per the fuzzing instructions [here](https://github.com/openssl/openssl/tree/master/fuzz) with the addition of `-fsanitize=integer -fsanitize-address-use-after-scope -fsanitize=float-divide-by-zero -fsanitize=float-cast-overflow`.

Running any of the fuzzers results in the following runtime error:

```
ssl/s3_lib.c:2796:19: runtime error: unsigned integer overflow: 50331649 - 50331650 cannot be represented in type 'unsigned int'
    #0 0x5b64ec in cipher_compare /root/openssl/ssl/s3_lib.c:2796:19
    #1 0x7f8bed725af9  (/lib/x86_64-linux-gnu/libc.so.6+0x34af9)
    #2 0x7f8bed725a51  (/lib/x86_64-linux-gnu/libc.so.6+0x34a51)
    #3 0x7f8bed725a51  (/lib/x86_64-linux-gnu/libc.so.6+0x34a51)
    #4 0x7f8bed725a51  (/lib/x86_64-linux-gnu/libc.so.6+0x34a51)
    #5 0x7f8bed725a51  (/lib/x86_64-linux-gnu/libc.so.6+0x34a51)
    #6 0x7f8bed725a51  (/lib/x86_64-linux-gnu/libc.so.6+0x34a51)
    #7 0x7f8bed725a51  (/lib/x86_64-linux-gnu/libc.so.6+0x34a51)
    #8 0x7f8bed725e39 in qsort_r (/lib/x86_64-linux-gnu/libc.so.6+0x34e39)
    #9 0x5b6184 in ssl_sort_cipher_list /root/openssl/ssl/s3_lib.c:2801:5
    #10 0x5feda5 in ssl_load_ciphers /root/openssl/ssl/ssl_ciph.c:350:5
    #11 0x621c4b in ossl_init_ssl_base /root/openssl/ssl/ssl_init.c:98:10
    #12 0x6219b4 in ossl_init_ssl_base_ossl_ /root/openssl/ssl/ssl_init.c:24:1
    #13 0x7f8bedebd738 in __pthread_once_slow (/lib/x86_64-linux-gnu/libpthread.so.0+0xe738)
    #14 0x10c1e17 in CRYPTO_THREAD_run_once /root/openssl/crypto/threads_pthread.c:107:9
    #15 0x6216be in OPENSSL_init_ssl /root/openssl/ssl/ssl_init.c:198:10
    #16 0x4ed864 in FuzzerInitialize /root/openssl/fuzz/server.c:496:5
    #17 0x4ed66e in LLVMFuzzerInitialize /root/openssl/fuzz/driver.c:20:12
    #18 0x4f18e9 in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) /root/test/Fuzzer/FuzzerDriver.cpp:527:5
    #19 0x4eeaf0 in main /root/test/Fuzzer/FuzzerMain.cpp:20:10
    #20 0x7f8bed7112b0 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x202b0)
    #21 0x420259 in _start (/root/openssl/fuzz/server+0x420259)

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ssl/s3_lib.c:2796:19
```

This issue is only visible if you compile OpenSSL with UBSan and `-fsanitize=integer`.

**Comment by levitte** [comment]
#3890 (fix similar to #3887) should fix this.

**Comment by geeknik** [comment]
I was close on the patch myself but I had `>` instead of `<` on line 2798.  Just re-compiled and I can verify #3890 fixes this. Thanks!
